pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -821,19 +717,15 @@ public void removeSegment(IndexSegment segment) {
       _logger.info("Skip removing segment: {} because metadata manager is already stopped", segmentName);
       return;
     }
-    // Skip removing the upsert metadata of segment that has max comparison value smaller than
-    // (largestSeenComparisonValue - TTL), i.e. out of metadata TTL. The expired metadata is removed while creating
-    // new consuming segment in batches.
-    boolean skipRemoveMetadata = false;
-    if (isOutOfMetadataTTL(segment)) {
-      _logger.info("Skip removing segment: {} because it's out of TTL", segmentName);
-      skipRemoveMetadata = true;
-    }
```

Comment:
this is a quick fix not quite related to the refactoring on preloading. Basically, move the check into try-catch in case isOutOfMetadataTTL throws exception, which should be rare but just in case.